### PR TITLE
feat: modify messages to be flowing through sync-widget and update props

### PR DESCRIPTION
**Pull request by benox3**

### CODE REVIEW

File: packages/js-sync-with-portabl/src/lib/createPostMessageSendClient.ts [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+  } = {
+    targetOrigin: '*',
+  },
+) {
+  const messageQueue: IMessageQueueItem<TMessageType>[] = [];
+  let isIframeLoaded = false;
+
+  const sendMessage = (message: TMessageType, targetOrigin?: string) => {
+    if (
+      isIframeLoaded &&
```
Comment (bao-portabl) [inline packages/js-sync-with-portabl/src/lib/createPostMessageSendClient.ts]: We can simplify this using optional chaining

```.ts
if (isIframeLoaded && iframeEl?.contentWindow?.postMessage) { 
    etc.. 
}
```

File: packages/js-sync-with-portabl/src/lib/createPostMessageSendClient.ts [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+  message: TMessageType;
+  targetOrigin: string;
+}
+
+export function createPostMessageSenderClient<
+  TMessageType extends IPostMessage,
+>(
+  iframeEl: HTMLIFrameElement,
+  {
+    targetOrigin: clientTargetOrigin = '*',
```
Comment (bao-portabl) [inline packages/js-sync-with-portabl/src/lib/createPostMessageSendClient.ts]: This is a bit hard to read with all the destructuring and typings. Can we make this pass in an options and have an interface like `PostMessageOptions`?

```  
options: PostMessageOptions = {targetOrigin: '*'}
```

File: packages/js-sync-with-portabl/src/lib/createPostMessageSendClient.ts [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+  }: {
+    targetOrigin?: string;
+  } = {
+    targetOrigin: '*',
+  },
+) {
+  const messageQueue: IMessageQueueItem<TMessageType>[] = [];
+  let isIframeLoaded = false;
+
```
Comment (bao-portabl) [inline packages/js-sync-with-portabl/src/lib/createPostMessageSendClient.ts]: Would probably need to set `clientTargetOrigin` as a var here using the updated interface

File: packages/js-sync-with-portabl/src/lib/constants.ts [hunk 4]
```diff
@@ -1,16 +1,15 @@
...
-export const SYNC_ACK = 'sync:acked';
-export const SYNC_REQUEST_ACK = 'sync:request-ack';
-export const SYNC_REQUEST_ERROR = 'sync:request-error';
-export const SYNC_PASSPORT_READY = 'sync:passport-ready';
-export const SYNC_PASSPORT_CLOSED = 'sync:passport-closed';
+export enum OutgoingPostMessageEvent {
+  SYNC_WIDGET_CONTEXT_LOADED = 'SYNC_WIDGET_CONTEXT_LOADED',
+  SYNC_WIDGET_ERROR = 'SYNC_WIDGET_ERROR',
+  SYNC_INVITATION_CREATED = 'SYNC_INVITATION_CREATED',
+  SYNC_INVITATION_ERROR = 'SYNC_INVITATION_CREATED',
```
Comment (bao-portabl) [inline packages/js-sync-with-portabl/src/lib/constants.ts]: Typo here, should be `SYNC_INVITATION_ERROR = 'SYNC_INVITATION_ERROR'`

File: packages/js-sync-with-portabl/src/index.ts [hunk 5]
```diff
@@ -1,182 +1,144 @@
...
+      iframeWidget,
+      { targetOrigin: widgetBaseUrl },
+    );
 
-    iframeWidget.contentWindow?.postMessage(
-      {
-        action: PREREQS_FAILED,
+  try {
+    const syncContext = await withRetries(getSyncContext, MAX_RETRIES);
+    const { isSessionEstablished, isSyncOn, datapoints, issuerDIDs } =
```
Comment (bao-portabl) [inline packages/js-sync-with-portabl/src/index.ts:46]: What is `issuerDIDs` used for?

File: packages/js-sync-with-portabl/src/index.ts [hunk 6]
```diff
@@ -1,182 +1,144 @@
...
 export async function createSyncWithPortabl(options: Options): Promise<void> {
   const {
     widgetBaseUrl = 'https://widgets.getportabl.com',
-    getPrereqs,
-    onUserConsent,
-    rootSelector,
+    getSyncContext,
+    prepareSync,
+    rootSelector = DEFAULT_ROOT_SELECTOR,
     // providerName,
```
Comment (bao-portabl) [inline packages/js-sync-with-portabl/src/index.ts]: is `providerName` still needed?

File: packages/js-sync-with-portabl/src/index.ts (hunk 5 above)
Comment (benox3) [reply packages/js-sync-with-portabl/src/index.ts:46]: we don't use it currently, but we may have reason to. issuerDIDs are the did of the provider.

File: packages/js-sync-with-portabl/src/lib/createPostMessageSendClient.ts (hunk 2 above)
Comment (benox3) [reply packages/js-sync-with-portabl/src/lib/createPostMessageSendClient.ts]: Good suggestion. will update.

File: packages/js-sync-with-portabl/src/index.ts (hunk 6 above)
Comment (benox3) [reply packages/js-sync-with-portabl/src/index.ts]: Yup added it back - we removed it temporarily.

**Review by bao-portabl — APPROVED** [review]
lgtm

**Pull request merged**